src/OptOut.php

```diff
@@ -0,0 +1,146 @@
+  public static function admin_init() {
+    // Register opt-out related settings.
+    register_setting(Plugin::PREFIX . '-settings', 'core_standards_optout_ids');
+    register_setting(Plugin::PREFIX . '-settings', 'core_standards_optout_script');
+    register_setting(Plugin::PREFIX . '-settings', 'core_standards_optout_active');
+
+    add_action('admin_enqueue_scripts', __CLASS__ . '::admin_enqueue_scripts');
+  }
+
+  /**
+   * @implements init
+   */
+  public static function init() {
+    if (is_admin()) {
+      return;
+    }
+
+    // Injects the Google Analytics IDs and the opt-out script in the page head
+    // section before any tracking script.
+    add_action('wp_head', __CLASS__ . '::wp_head', 5);
+
+    // Register the shortcode to call the opt-out cookie creation script.
+    add_shortcode('ga-optout-link', __CLASS__ . '::registerOptOutShortcode');
+
+    // Enqueue frontend scripts.
+    add_action('wp_enqueue_scripts', __CLASS__ . '::wp_enqueue_scripts');
+  }
+
+  /**
+   * @implements wp_head
+   */
+  public static function wp_head() {
+    if (!get_option('core_standards_optout_active')) {
+      return;
+    }
+    // Converts any non-word character into a comma and removes white-space characters
+    // (except '-') from the string to get a comma-separated list of Google Analytics IDs.
+    $optout_ids = preg_replace(['@\W^-|\n@', '@\s@'], [',', ''], esc_attr(get_option('core_standards_optout_ids')));
+  ?>
+
+<script>
+// List of Google Analytics IDs.
+window.gaIdList = <?= json_encode(explode(',', $optout_ids)) ?>;
+</script>
+  <?php
+    echo get_option('core_standards_optout_script') ?: static::$opt_out_script;
+  }
+
+  /**
+   * Renders opt-out link shortcode.
+   *
+   * @param array $atts Shortcode parameters as attributes.
+   * @param string $content Shortcode wrapped content.
+   * @return string
+   */
+  public static function registerOptOutShortcode ($atts, $content = NULL) {
+    if (get_option('core_standards_optout_active')) {
+      return '<a class="core-standards-optout-link" href="#">' . $content . '</a>';
+    }
+    else {
+      return $content;
+    }
+  }
+
+  /**
+   * Creates a submenu page in the WordPress admin "Settings" section.
+   *
+   * @implements admin_menu
+   */
+  public static function admin_menu() {
+    add_submenu_page('options-general.php', __('WordPress Core Standards Settings', Plugin::L10N), __('WordPress Core Standards', Plugin::L10N), 'manage_options', 'core-standards-settings', [__CLASS__, 'renderSettings']);
+  }
+
+  /**
+   * Renders the settings page template.
+   */
+  public static function renderSettings() {
+    Plugin::renderTemplate(['templates/settings.php']);
+  }
+
+  /**
+   * @implements wp_enqueue_scripts.
+   */
+  public static function wp_enqueue_scripts() {
+    if (get_option('core_standards_optout_active')) {
+      wp_enqueue_script(Plugin::PREFIX . '-opt-out', Plugin::getBaseUrl() . '/assets/scripts/opt-out.js', ['jquery'], FALSE, TRUE);
```

Comment:
The compiled/optimized scripts from dist should be loaded instead of the source scripts from assets.

Also note that the compiled script files in dist are missing in this PR.

Reply:
@sun There is no build pipeline set for core-standards plugin (I found this when I started implementing). If I shall set it up I will need to correct my initial time estimation.